pandas/tests/io/parser/test_parsers.py

```diff
@@ -25,6 +28,18 @@
 from .dtypes import DtypeTests
 
 
+@pytest.mark.high_memory
+def test_bytes_exceed_2gb():
```

Comment:
I would refactor to add this into the `CParser` tests.  You don't necessarily have to have the test run for the low-memory c-parser (you can just have it skip if it is the case).